### DISCUSSION TITLE
can MOOSE do Multi-point constraints (MPCs)?

### DISCUSSION THREAD

**Discussion by Traiwit** [idaholab/moose#18342] [Q&A General] [answered]
Hi guys,

I just wondering if MOOSE can perform Multi-point constraints (MPCs)?

from a quick search, I see MOOSE has `EqualValueEmbeddedConstraint `https://mooseframework.inl.gov/moose/source/constraints/EqualValueEmbeddedConstraint.html

But this might not be what I am looking for

I want something that can match 2 DOFs as follow

![1](https://user-images.githubusercontent.com/79562439/125745834-cd058dcb-40d4-46f5-8866-2d9d9682295a.PNG)

Thanks guys

Kind regards,
Traiwit

**Comment by Traiwit** [comment]
I think `LinearNodalConstraint` is the one
https://mooseframework.inl.gov/source/constraints/LinearNodalConstraint.html

Does anyone know how to make a variable of node A = node B?

what should I set the value of `penalty` and `weights` to?

**Comment by hugary1995** [comment, marked as answer]
If you are asking "whether moose can do MPC", the answer is definitely yes -- the infrastructure is there.

But I'm not sure if there are already some classes implemented for MPC. I think you are in the right direction. Classes like `LinearNodalConstraint` and `EqualValueBoundaryConstraint` can be a good starting point if you want to implement MPC in your own application.
